src/condor_tests/test_curl_structured_error.py

```diff
@@ -0,0 +1,275 @@
+    time.sleep(32)
+    return Response()
+
+
+@action
+def the_open_port():
+    server = HTTPServer()
+    if not server.is_running():
+        server.start()
+
+    # Arguably, all of these kind of lines throughout this whole test
+    # should be in the corresponding test case, instead.
+    server.expect_request("/400").respond_with_data("", status=400)
+    server.expect_request("/403").respond_with_data("", status=403)
+    server.expect_request("/404").respond_with_data("", status=404)
+
+
+    yield server.port
+
+    server.stop()
+
+
+@action
+def the_slow_port():
+    server = HTTPServer()
+    if not server.is_running():
+        server.start()
+
+    # This must be its own server, because otherwise the order of the requests
+    # matter, since each server is single-threaded.
+    server.expect_request("/slow").respond_with_handler(slow_reply_handler)
+
+    yield server.port
+
+    server.stop()
+
+
+@action
+def the_redirect_port():
+    server = HTTPServer()
+    if not server.is_running():
+        server.start()
+
+    # Joyously, curl doesn't respond to this consistently: some of the time
+    # it reports error 78 (CURLE_REMOTE_FILE_NOT_FOUND) and some of the time
+    # it reports error 47 (CURLE_TOO_MANY_REDIRECTS).  (This is assuming the
+    # plug-in has been modified to set CURLOPT_MAXREDIRS, which doesn't have
+    # a default value in older versions of libcurl.)
+
+    server.expect_request("/tmr").respond_with_data("", status=302,
+        headers=[{"Location", f"http://localhost:{server.port}/tms"}]
+    )
+    server.expect_request("/tms").respond_with_data("", status=302,
+        headers=[{"Location", f"http://localhost:{server.port}/tmr"}]
+    )
+
+    yield server.port
+
+    server.stop()
+
+
+@action
+def the_job_handles(test_dir, the_condor, the_job_description, the_closed_port, the_open_port, the_slow_port, the_redirect_port):
+    job_handles = {}
+
+    for name, test_case in TEST_CASES.items():
+        complete_job_description = {
+            ** the_job_description,
+            "transfer_input_files": test_case["url"].format(
+                closed_port=the_closed_port,
+                open_port=the_open_port,
+                slow_port=the_slow_port,
+                redirect_port=the_redirect_port,
+            ),
+            "log": (test_dir / f"{name}.log").as_posix(),
+        }
+        job_handle = the_condor.submit(
+            description = complete_job_description,
+            count = 1
+        )
+
+        job_handles[name] = job_handle
+
+    yield job_handles
+
+
+@action(params={name: name for name in TEST_CASES})
+def test_case_pair(request, the_job_handles):
+    return (request.param, the_job_handles[request.param])
+
+
+@action
+def test_name(test_case_pair):
+    return test_case_pair[0]
+
+
+@action
+def test_job(test_case_pair):
+    return test_case_pair[1]
+
+
+@action
+def test_case(test_case_pair):
+    return TEST_CASES[test_case_pair[0]]
```

Comment:
Could've just passed down `test_name` instead of `test_case_pair`. Not saying you should change this just wanted to point it out

Reply:
Good point.  Yeah, that probably would have been clearer.